src/Swarm/TUI/Launch/Prep.hs

```diff
@@ -51,11 +52,14 @@ parseSeedInput seedEditor =
  where
   seedFieldText = mconcat $ getEditContents seedEditor
 
-parseWidgetParams :: LaunchControls -> IO EditingLaunchParams
-parseWidgetParams (LaunchControls (FileBrowserControl _fb maybeSelectedScript _) seedEditor _ _) = do
+parseWidgetParams :: Maybe ProcessedTerm -> LaunchControls -> IO EditingLaunchParams
+parseWidgetParams solution (LaunchControls autoPlay (FileBrowserControl _fb maybeSelectedScript _) seedEditor _ _) = do
   eitherParsedCode <-
-    runThrow . withThrow (prettyText @SystemFailure) $
-      traverse parseCodeFile maybeSelectedScript
+    if autoPlay
+      then pure . Right $ CodeToRun ScenarioSuggested <$> solution
+      else
+        runThrow . withThrow (prettyText @SystemFailure) $
+          traverse parseCodeFile maybeSelectedScript
```

Comment:
I tried to get current scenario solution here, but I am not sure this path is even run. 😕

I attempted to debug it using `hPutStrLn stderr`, but did not see either branch print anything.

Maybe it's just something crazy like in #1456 and I need to test it in a clean docker container.

Reply:
OK, so it's only run if script is selected.

I will need to look around and see why.